### DISCUSSION TITLE
Welcome to grizzly Discussions!

### DISCUSSION THREAD

**Discussion by malcolmholmes** [grafana/grizzly#103] [General]
## 👋 Welcome!

Grizzly is a tool in which we are exploring ideas about how to interact with various observability resources, "from code".

> :warning: Grizzly is currently alpha code - the APIs etc that it exposes can, and likely will change.

This Discussion board is here to allow you to ask questions and explore possible usages and features. We can also use it to share new features.

**Comment by DamGameHero** [comment]
Hello !  

What do you think about using the [provisioning the folder structure from filesystem to Grafana](https://grafana.com/docs/grafana/latest/administration/provisioning/#provision-folders-structure-from-filesystem-to-grafana) via a `grr export` ?

Currently with `grr export` in last release ([v 0.1.0](https://github.com/grafana/grizzly/releases/tag/v0.1.0)), dashboards are created in: <pre>my-provisioning-dir/<b>Dashboard</b>/dashboard.json</pre>`Dashboard` folder is a value that can't be changed. Line here, I guess: https://github.com/grafana/grizzly/blob/b719359dca83a1ea0f1bef77aec6bd1ae02cb1ae/pkg/grizzly/workflow.go#L387

However, to take advantage of the folder structure, it would be preferable to build the path like this:<pre>my-provisioning-dir/<b>grafanaDashboardFolder</b>/dashboard.json</pre> With `grafanaDashboardFolder` given like in the `mydash.libsonnet` [example](https://github.com/grafana/grizzly#grafana-dashboard-example) or in another way .

I had done this feature, but it was on the previous version. This has evolved since and I would like to know  your opinion on this point.

**Reply by malcolmholmes** [reply]
Hi, firstly, welcome! Secondly, if poss, it is better to create a seperate thread. Will help others find this in future.

To answer your question, I am working on a next version, and your feedback is very useful. I want to make Grizzly able to pull from Grafana, so pull raw json and write it (likely as yaml manifests) to disk. For this I need a way to specify a directory structure. This fits well with your need.

It is also useful to be reminded that file system provisioning requires raw unwrapped json blobs, which is slightly different to the route I was going.

I hope to make some PRs this weekend, and would appreciate comment. I don't expect the PRs to get merged immediately, they will be more about exploring APIs etc.

**Reply by DamGameHero** [reply]
Ok thanks for the reply. Seperate thread: https://github.com/grafana/grizzly/discussions/106
